.pre-commit-config.yaml

```diff
@@ -1,7 +1,7 @@
 default_language_version:
-  python: python3.9
+  python: python3.11
 
-default_stages: [commit, push]
+#default_stages: [commit, push]
```

Comment:
Since we're doing a clean-up, do we also want to update the repo revs? add a few more pre-commit hooks (e.g., check-docstring-first, check-merge-conflict, check-toml, mixed-line-ending, trailing-whitespace)

Reply:
Cool yeah, I'll look through here and add some that aren't already covered by flake & mypy

https://github.com/pre-commit/pre-commit-hooks

Reply:
Could use ruff instead of flake8